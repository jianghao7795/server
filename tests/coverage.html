
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>tests: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">server-fiber/tests/test_helpers.go (0.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package tests

import (
        "bytes"
        "encoding/json"
        "net/http"
        "net/http/httptest"
        "testing"

        "github.com/gofiber/fiber/v2"
        "github.com/stretchr/testify/assert"
)

// TestHelper 提供测试辅助功能
type TestHelper struct{}

// NewTestHelper 创建新的测试辅助实例
func NewTestHelper() *TestHelper <span class="cov0" title="0">{
        return &amp;TestHelper{}
}</span>

// CreateTestApp 创建用于测试的 Fiber 应用
func (h *TestHelper) CreateTestApp() *fiber.App <span class="cov0" title="0">{
        app := fiber.New()
        return app
}</span>

// AssertResponseStatus 验证响应状态码
func (h *TestHelper) AssertResponseStatus(t *testing.T, resp *httptest.ResponseRecorder, expectedStatus int) <span class="cov0" title="0">{
        assert.Equal(t, expectedStatus, resp.Code)
}</span>

// AssertResponseBody 验证响应体结构
func (h *TestHelper) AssertResponseBody(t *testing.T, resp *httptest.ResponseRecorder, expectedKeys ...string) <span class="cov0" title="0">{
        var responseBody map[string]interface{}
        err := json.NewDecoder(resp.Body).Decode(&amp;responseBody)
        assert.NoError(t, err)

        for _, key := range expectedKeys </span><span class="cov0" title="0">{
                assert.Contains(t, responseBody, key)
        }</span>
}

// AssertResponseMessage 验证响应消息
func (h *TestHelper) AssertResponseMessage(t *testing.T, resp *httptest.ResponseRecorder, expectedMessage string) <span class="cov0" title="0">{
        var responseBody map[string]interface{}
        err := json.NewDecoder(resp.Body).Decode(&amp;responseBody)
        assert.NoError(t, err)

        if msg, exists := responseBody["msg"]; exists </span><span class="cov0" title="0">{
                assert.Equal(t, expectedMessage, msg)
        }</span>
}

// CreateTestRequest 创建测试请求
func (h *TestHelper) CreateTestRequest(method, path string, body []byte) *http.Request <span class="cov0" title="0">{
        req := httptest.NewRequest(method, path, bytes.NewReader(body))
        if body != nil </span><span class="cov0" title="0">{
                req.Header.Set("Content-Type", "application/json")
        }</span>
        <span class="cov0" title="0">return req</span>
}

// MockHTTPClient 模拟 HTTP 客户端
type MockHTTPClient struct {
        Responses map[string]*MockResponse
}

type MockResponse struct {
        StatusCode int
        Body       []byte
        Error      error
}

func NewMockHTTPClient() *MockHTTPClient <span class="cov0" title="0">{
        return &amp;MockHTTPClient{
                Responses: make(map[string]*MockResponse),
        }
}</span>

func (m *MockHTTPClient) SetResponse(url string, response *MockResponse) <span class="cov0" title="0">{
        m.Responses[url] = response
}</span>

// TestData 提供测试用的示例数据
type TestData struct{}

func NewTestData() *TestData <span class="cov0" title="0">{
        return &amp;TestData{}
}</span>

// GetSampleGithubData 获取示例 GitHub 数据
func (td *TestData) GetSampleGithubData() []map[string]interface{} <span class="cov0" title="0">{
        return []map[string]interface{}{
                {
                        "author":     "testuser1",
                        "message":    "feat: add new feature",
                        "commitTime": "2024-01-01 12:00:00",
                },
                {
                        "author":     "testuser2",
                        "message":    "fix: resolve bug",
                        "commitTime": "2024-01-02 13:00:00",
                },
        }
}</span>

// GetSamplePageInfo 获取示例分页信息
func (td *TestData) GetSamplePageInfo() map[string]interface{} <span class="cov0" title="0">{
        return map[string]interface{}{
                "page":     1,
                "pageSize": 10,
        }
}</span>
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
