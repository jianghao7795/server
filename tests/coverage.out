mode: set
server-fiber/tests/test_helpers.go:18.34,20.2 1 0
server-fiber/tests/test_helpers.go:23.49,26.2 2 0
server-fiber/tests/test_helpers.go:29.110,31.2 1 0
server-fiber/tests/test_helpers.go:34.112,39.35 4 0
server-fiber/tests/test_helpers.go:39.35,41.3 1 0
server-fiber/tests/test_helpers.go:45.115,50.48 4 0
server-fiber/tests/test_helpers.go:50.48,52.3 1 0
server-fiber/tests/test_helpers.go:56.88,58.17 2 0
server-fiber/tests/test_helpers.go:58.17,60.3 1 0
server-fiber/tests/test_helpers.go:61.2,61.12 1 0
server-fiber/tests/test_helpers.go:75.42,79.2 1 0
server-fiber/tests/test_helpers.go:81.74,83.2 1 0
server-fiber/tests/test_helpers.go:88.30,90.2 1 0
server-fiber/tests/test_helpers.go:93.68,106.2 1 0
server-fiber/tests/test_helpers.go:109.64,114.2 1 0
